### ISSUE TITLE
Action fails to upload source to bucket, function deployment silently fails.

### ISSUE THREAD

**Issue by uschtwill** [google-github-actions/deploy-cloud-functions#26]
### TL;DR

The action intermittently breaks the function deployment by failing to upload the source code.

**Expected behavior**

To work reliably.

**Observed behavior**

Sometimes the action fails to upload the zip file to the Cloud Storage bucket. The function deployment then fails with the following message:
```
Build failed: could not resolve storage source: googleapi: Error 404: No such object: gcf-sources-856243217562-europe-west3/function-9c18cebc-b919-4ef3-bf9f-edfde82cf6cd/version-25/function-source.zip, notFound
```

### Reproduction
I tried to isolate the behaviour and to consistently reproduce it, but failed. Initially I thought it would happen, when re-running jobs from the jobs overview in the Github UI or when the commit's message is the same as the previous commit's message. But now it just happened 3 times back to back with "normal" commits, and then stopped happening on a 4th "normal" commit. Almost feels like a caching bug ...fleeting.

So I gave up. As everything I changed in-between commits was to increment a commented number in the .yaml, I really don't know what's happening here. The action always pretends to have been successful and doesn't show any errors, no matter what. The only give-away in the Github UI and the job's logs is that the action only takes a couple of seconds, if the bug happens, but considerably longer, if it does not.

During the build we pull private packages from Github and the repo is a lerna-managed mono-repo (see "source_dir"). Apart from that I can't think of anything "special" about our setup.

**Action YAML**

```yaml
      - name: Deploy Cloud Function
        uses: google-github-actions/deploy-cloud-functions@main
        with:
          name: function
          runtime: nodejs12
          region: europe-west3
          source_dir: packages/function-package
          entry_point: function
```

**Screenshots**
*****
<img width="1133" alt="Screenshot 2021-01-14 at 18 53 33" src="https://user-images.githubusercontent.com/10156948/104631222-41936e00-569c-11eb-92c2-eec2bec084fc.png">

*****

<img width="1463" alt="Screenshot 2021-01-14 at 18 51 50" src="https://user-images.githubusercontent.com/10156948/104631235-448e5e80-569c-11eb-86ff-732201556c1a.png">

*****

<img width="1149" alt="Screenshot 2021-01-14 at 20 45 18" src="https://user-images.githubusercontent.com/10156948/104641264-d3ee3e80-56a9-11eb-9dcc-2303efe87adb.png">

**Comment by bharathkkb** [comment]
Hi @uschtwill 
Thanks for the report. I did see a possible edge case where it could fail and will improve the error handling incase of failure.

**Comment by uschtwill** [comment]
Hey @bharathkkb
Thanks! This will indeed greatly improve the situation in regards to observability. Although I am afraid the root cause will remain untouched. Do you have any ideas on what could cause this in the first place?

**Comment by bharathkkb** [comment]
I believe the root cause may have been a race condition as we were not awaiting the upload request completion. Hopefully the PR fixes it or sheds more light if it is another cause.
